core/src/main/scala/cats/data/Cokleisli.scala

```diff
@@ -11,9 +11,31 @@ import scala.annotation.tailrec
  */
 final case class Cokleisli[F[_], A, B](run: F[A] => B) { self =>
 
+  /**
+    * Example:
+    * {{{
+    * scala> import cats.Id, cats.implicits._
+    * scala> val x : Id[Int] = 42
+    * scala> def before(x: Int) = x + 1
+    * scala> def after(x: Int) = x - 1
+    * scala> val example : Cokleisli[Id,Int,Int] = Cokleisli((f: Id[Int]) => f.extract)
+    * scala> example.dimap(before)(after) == 42
```

Comment:
Can you add what this returns here and below?

Reply:
Thanks @LukaJCB for the reminder, appreciate it.

Reply:
@johnynek thanks for the review and agree with you. Following on your lead, i've revised the examples to use `NonEmptyList` instead, let me know what you think when you look at them?